OpenEdXMobile/src/main/java/org/edx/mobile/base/BaseFragment.java

```diff
@@ -33,4 +47,40 @@ public void onResume() {
      */
     protected void onRevisit() {
     }
+
+    /**
+     * This method is used when app needs to check permission(s) and if permission(s) is allowed then listener
+     * method trigger otherwise in else part show popup for requesting permission then user allow or
+     * deny that permission(s) and result return in onRequestPermissionsResult method
+     *
+     * @param permissions Permissions required to use mobile resource
+     * @param requestCode unique request code for onRequestPermissionsResult method to diffrentiate different
+     *                    permissions results
+     */
+    protected void askForPermission(String[] permissions, int requestCode) {
```

Comment:
Following would be a better javadoc:

```java
    /**
     * Checks the status of the provided permissions, if a permission has been given, a callback
     * function is called otherwise the said permission is requested.
     *
     * @param permissions The requested permissions.
     * @param requestCode The request code passed in {@link #requestPermissions(String[], int)}.
     */
```